src/test/poll/test_poll.rs

```diff
@@ -0,0 +1,223 @@
+/*
+ * The Shadow Simulator
+ * See LICENSE for licensing information
+ */
+
+use test_utils::set;
+use test_utils::TestEnvironment as TestEnv;
+
+fn fd_write(fd: i32) -> i32 {
+    return unsafe { libc::write(fd, b"test".as_ptr() as *const libc::c_void, 4) as i32 };
+}
+
+fn fd_read_cmp(fd: i32) -> Result<(), String> {
+    let mut buf = [0_u8; 4];
+    nix::unistd::read(fd, &mut buf).map_err(|e| e.to_string())?;
+    println!("read from buf: {:?}", buf);
+    if &buf != b"test" {
+        return Err("error: did not read 'test' from pipe.".to_string());
+    } else {
+        return Ok(());
+    }
+}
+
+fn test_pipe() -> Result<(), String> {
+    /* Create a set of pipefds
+    pfd[0] == read, pfd[1] == write */
+    let mut pfds = [0_i32; 2];
+    if unsafe { libc::pipe(pfds.as_mut_ptr() as *mut libc::c_int) < 0 } {
+        return Err("error: pipe could not be created!".to_string());
+    }
+
+    /* poll will check when pipe has info to read */
+    let mut read_poll = libc::pollfd {
+        fd: pfds[0],
+        events: libc::POLLIN,
+        revents: 0,
+    };
+
+    /* First make sure there's nothing there */
+    let mut ready = unsafe { libc::poll(&mut read_poll as *mut libc::pollfd, 1, 100) };
```

Comment:
Everything in the function at this point can be wrapped with `test_utils::run_and_close_fds`, which will automatically close the file descriptors when we exit the closure and lets you remove all of the `libc::close()` calls. This helps make the code more readable and prevents us from forgetting to close file descriptors:

```rust
test_utils::run_and_close_fds(&[pfd_read, pfd_write], || {
    let mut read_poll = ...
    let mut ready = ...
    if ready < 0 {
        return Err("error: poll failed".to_string());
    } else if ready > 0 {
        return Err(...);
    }
    fd_write(pfds[1])?;
    ...
})?;
```

For example see: https://github.com/shadow/shadow/blob/5079bb55ecc7ad7574389fadfe0d6eed7f56fcf9/src/test/socket/sendto_recvfrom/test_sendto_recvfrom.rs#L322-L342